pymtl3/passes/sim/PrepareSimPass.py

```diff
@@ -0,0 +1,298 @@
+"""
+========================================================================
+AddSimUtilFuncsPass.py
+========================================================================
+
+Author : Shunning Jiang
+Date   : Jan 26, 2020
+"""
+
+import py
+
+from pymtl3.datatypes import Bits, b1
+from pymtl3.dsl.Component import Component
+from pymtl3.dsl.Connectable import Const, Interface, MethodPort, Signal
+from pymtl3.dsl.NamedObject import NamedObject
+from pymtl3.extra.pypy import custom_exec
+from pymtl3.passes.backends.verilog import VerilogTBGenPass
+from pymtl3.passes.BasePass import BasePass, PassMetadata
+from pymtl3.passes.errors import PassOrderError
+from pymtl3.passes.tracing.CLLineTracePass import CLLineTracePass
+from pymtl3.passes.tracing.LineTraceParamPass import LineTraceParamPass
+from pymtl3.passes.tracing.PrintTextWavePass import PrintTextWavePass
+from pymtl3.passes.tracing.VcdGenerationPass import VcdGenerationPass
+
+from .SimpleTickPass import SimpleTickPass
+
+
+class PrepareSimPass( BasePass ):
+  def __init__( self, print_line_trace=True, reset_active_high=True ):
+    assert reset_active_high in [ True, False ]
+
+    self.print_line_trace  = print_line_trace
+    self.reset_active_high = reset_active_high
+
+  def __call__( self, top ):
+    if hasattr(top, "sim_reset"):
+      raise AttributeError( "Please rename the attribute top.sim_reset")
+    if hasattr(top, "print_line_trace"):
```

Comment:
Have we decided to rename `line_trace` to `print_line_trace`?

Reply:
I haven't changed the line_trace convention. The print_line_trace function is a generated function that calls line_trace .